pinot-common/src/main/java/org/apache/pinot/pql/parsers/pql2/ast/InPredicateAstNode.java

```diff
@@ -108,7 +111,29 @@ public FilterQueryTree buildFilterQueryTree() {
       filterOperator = FilterOperator.IN;
     }
 
-    return new FilterQueryTree(_identifier, new ArrayList<>(values), filterOperator, null);
+    return new FilterQueryTree(_identifier, values, filterOperator, null);
+  }
+
+  @Override
+  public Expression buildFilterExpression() {
+    if (_identifier == null) {
+      throw new Pql2CompilationException("IN predicate has no identifier");
+    }
+    FilterKind filterOperator;
+    if (_isNotInClause) {
+      filterOperator = FilterKind.NOT_IN;
+    } else {
+      filterOperator = FilterKind.IN;
+    }
+    Expression expr = RequestUtils.getFunctionExpression(filterOperator.name());
+    expr.getFunctionCall().addToOperands(RequestUtils.getIdentifierExpression(_identifier));
+    for (AstNode astNode : getChildren()) {
+      if (astNode instanceof LiteralAstNode) {
```

Comment:
Should we dedup here?

Reply:
Function operands should not be dedeuped

Reply:
The behavior for this and filterQueryTree are different

Reply:
fixed. added test case